components/buildless-serverless/internal/controller/git/git_auth.go

```diff
@@ -0,0 +1,240 @@
+}
+
+func (a *GitAuth) loadSecret(ctx context.Context) error {
+	s := &corev1.Secret{}
+	err := a.client.Get(ctx,
+		types.NamespacedName{
+			Namespace: a.secretNamespace,
+			Name:      a.secretName,
+		}, s)
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to get secret: %s", err.Error()))
+	}
+	a.secret = s
+	return nil
+}
+
+func (a *GitAuth) parseSecret() error {
+	switch a.secret.Type {
+	case corev1.SecretTypeSSHAuth:
+		return a.parseSshAuthKubernetesSecret()
+	case corev1.SecretTypeBasicAuth:
+		return a.parseBasicAuthKubernetesSecret()
+		// It is for compatibility with the previous implementation
+	default:
+		switch a.authType {
+		case serverlessv1alpha2.RepositoryAuthSSHKey:
+			return a.parseSshAuthOldServerlessSecret()
+		case serverlessv1alpha2.RepositoryAuthBasic:
+			return a.parseBasicAuthOldServerlessSecret()
+		default:
+			return errors.New("unexpected authorization type")
+		}
+	}
+}
+
+func (a *GitAuth) GetAuthMethod() (transport.AuthMethod, error) {
+	switch a.authType {
+	case serverlessv1alpha2.RepositoryAuthSSHKey:
+		return a.sshAuth()
+	case serverlessv1alpha2.RepositoryAuthBasic:
+		return a.basicAuth()
+	default:
+		return nil, errors.New("unexpected authorization type")
+	}
+}
+
+func (a *GitAuth) GetAuthEnvs() []corev1.EnvVar {
+	s := a.secretName
+	var envs []corev1.EnvVar
+	envs = append(envs, corev1.EnvVar{
+		Name:  repositoryAuthTypeEnvVarName,
+		Value: string(a.authType),
+	})
+	envs = addEnvVar(envs, a.sshKey, s)
+	envs = addEnvVar(envs, a.username, s)
+	envs = addEnvVar(envs, a.password, s)
+	return envs
+}
+
+const (
+	kubernetesKeyFieldName         = "ssh-privatekey"
+	kubernetesUsernameFieldName    = "username"
+	kubernetesPasswordFieldName    = "password"
+	oldServerlessKeyFieldName      = "key"
+	oldServerlessUsernameFieldName = "username"
+	oldServerlessPasswordFieldName = "password"
+	repositoryAuthTypeEnvVarName   = "APP_REPOSITORY_AUTH_TYPE"
+	usernameEnvVarName             = "APP_REPOSITORY_USERNAME"
+	passwordEnvVarName             = "APP_REPOSITORY_PASSWORD"
+	sshKeyEnvVarName               = "APP_REPOSITORY_KEY"
+)
+
+func (a *GitAuth) parseSshAuthKubernetesSecret() error {
+	if a.authType != serverlessv1alpha2.RepositoryAuthSSHKey {
+		return errors.New(fmt.Sprintf("inconsistent secret type: %s, %s", a.authType, corev1.SecretTypeSSHAuth))
+	}
+	privateKey, ok := a.secret.Data[kubernetesKeyFieldName]
+	if !ok {
+		return errors.New(fmt.Sprintf("missing '%s'", kubernetesKeyFieldName))
+	}
+	a.sshKey = &dataField[[]byte]{
+		value:     privateKey,
+		fieldName: kubernetesKeyFieldName,
+		envName:   sshKeyEnvVarName,
+	}
+	return nil
+}
+
+func (a *GitAuth) parseBasicAuthKubernetesSecret() error {
+	if a.authType != serverlessv1alpha2.RepositoryAuthBasic {
+		return errors.New(fmt.Sprintf("inconsistent secret type: %s, %s", a.authType, corev1.SecretTypeBasicAuth))
+	}
+	username, usernameFound := a.secret.Data["username"]
+	password, passwordFound := a.secret.Data["password"]
```

Comment:
Why do you decide to not use the`kubernetesUsernameFieldName` and `kubernetesPasswordFieldName` variables here?

Reply:
fixed